src/docs/arch-row.dox

```diff
@@ -1,12 +1,68 @@
-/*! @arch_page arch-row Row Store
+/*! @arch_page arch-row-column Row Store & Column Store
 
-Row Stores are Btrees stored in WiredTiger that do not have a record id
-as key.  Thus, they implement a generalized version of a Btree, where
-the key and data can be arbitrary length.
+@section Definition
+
+A row oriented storage is a database management system where all columns of a row are stored
+together, the data is stored by row, while a column oriented system stores its data tables by
+column, the data is stored by column. WriedTiger supports both and each management system is
+represented by a certain type of B-Tree data structure.
+
+Keys and values in row-stores can be anything of arbitrary length, thus, row-stores implement a
+generalized version of a B-Tree. However, column-stores use a 64-bit unsigned integer record id as
+keys, thus, column-stores implement a specialized version of a B-Tree where the key has a
+predictable length.
+
+There are two types of column-stores in WiredTiger:
+- The fixed length column-store, as its name implies, the value has a fixed length, and furthermore
+the value is restricted to 1 and 8 bits in length. The bit length is specified when the column store
+ is created. The fixed length column store has specialized use cases like bitmaps
+- The variable length is a more general case which allows for values to have any length, and may
+have arbitrary types, including aggregates of various types.
+
+@section Main differences
+
+Column-store keys are derived from the value's location in the table while row-store keys are
+physically stored in the table. This means there is no repetition of keys across multiple files for
+column-stores while row-store keys are duplicated usually within the same single file. Therefore,
+column-stores have a real benefit over row-stores when it comes to finding a record, since there is
+only one key per page and keys take almost no storage, fast lookups can be performed and it is easy
+to move between records. For row-stores, a slower approach is necessary (i.e multiple binary
+searches) to locate a particular record and ultimately pull together a complete row.
+
+When reading and writing column-stores, only the columns required for any particular query are
+maintained in memory. For row-stores, the memory usage will depend on the definition of column
+groups required by the query.
+
+Generally, row-stores are faster for queries where all of the columns are required by every lookup
+because there is only a single set of meta-data pages to read into the cache and search.
```

Comment:
"meta-data"? You mean internal?

Reply:
I took it from [this doc](https://github.com/wiredtiger/wiredtiger/blob/develop/src/docs/file-formats.dox#L15) as I thought it would be interesting to have this information here.
I will use "internal" if it is more appropriate, I am still learning the different terminologies, it did not sound wrong to me when I read it!

Reply:
Ah, got it -- I'd say that "internal" is a better word in both cases. Metadata has an additional meaning, the schema files for the database.